File: .NET/VS2010TrainingKit/Labs/10 - Using MVVM/Source/Completed/VB/CustomersService/ICustomerService.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System.ServiceModel
Imports System.ServiceModel.Web
Imports CustomersService.Model.Entities

Namespace CustomersService
	<ServiceContract>
	Public Interface ICustomerService
		<OperationContract>
		Function GetCustomers() As List(Of Customer)

		<OperationContract>
		Function SaveCustomer(ByVal customer As Customer) As OperationStatus

		<OperationContract>
		Function GetCustomer(ByVal custID As Integer) As Customer

	End Interface
End Namespace

File: .NET/VS2010TrainingKit/Labs/10 - Using MVVM/Source/Starting Point/C#/CustomersService/ICustomerService.cs
﻿// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.ServiceModel;
using System.ServiceModel.Web;
using CustomerService.Model;
using CustomersService.Model.Entities;

namespace CustomersService
{
    [ServiceContract]
    public interface ICustomerService
    {
        [OperationContract]
        List<Customer> GetCustomers();

        [OperationContract]
        OperationStatus SaveCustomer(Customer customer);

        [OperationContract]
        Customer GetCustomer(int custID);

    }
}